src/storage/page_buffer.c

```diff
@@ -10441,8 +10452,9 @@ pgbuf_is_valid_page (THREAD_ENTRY * thread_p, const VPID * vpid, bool no_error,
     {
       if (valid != DISK_ERROR && !no_error)
 	{
+	  const char *vlabel = fileio_get_volume_label (vpid->volid, PEEK);
```

Comment:
shouldn't this forking be done inside `fileio_get_volume_label`? there are many calls (both direct and indirect) to this function

Reply:
it cannot be forked inside because fileio_get_volume_label may be called on temporary volumes too. if called on permanent volumes, it will cause errors or (debug) crashes and that's fine.

Reply:
But is it correct to say that, on a TSRS, there are potentially more places where the call would return NULL? In total there seem to be ~40 calls to this function.
If yes, shouldn't those be amended as well?

Reply:
You are right, some of those 40 calls should be amended as well. It requires careful examination, so I think it may be handled as a separate issue and pull request. What do you think?